application/src/main/java/org/togetherjava/tjbot/commands/moderation/scam/ScamBlocker.java

```diff
@@ -0,0 +1,356 @@
+        reportScamMessage(event, "Is this scam? (already deleted)", createConfirmDialog(event));
+    }
+
+    private void takeActionAutoDeleteAndQuarantine(@NotNull GuildMessageReceivedEvent event) {
+        addScamToHistory(event);
+        logScamMessage(event);
+        deleteMessage(event);
+        quarantineAuthor(event);
+        dmUser(event);
+        reportScamMessage(event, "Detected and handled scam", null);
+    }
+
+    private void addScamToHistory(@NotNull GuildMessageReceivedEvent event) {
+        scamHistoryStore.addScam(event.getMessage(), MODES_WITH_IMMEDIATE_DELETION.contains(mode));
+    }
+
+    private void logScamMessage(@NotNull GuildMessageReceivedEvent event) {
+        logger.warn("Detected a scam message ('{}') from user '{}' in channel '{}' of guild '{}'.",
+                event.getMessageId(), event.getAuthor().getId(), event.getChannel().getId(),
+                event.getGuild().getId());
+    }
+
+    private void deleteMessage(@NotNull GuildMessageReceivedEvent event) {
+        event.getMessage().delete().queue();
+    }
+
+    private void quarantineAuthor(@NotNull GuildMessageReceivedEvent event) {
+        quarantineAuthor(event.getGuild(), event.getMember(), event.getJDA().getSelfUser());
+    }
+
+    private void quarantineAuthor(@NotNull Guild guild, @NotNull Member author,
+            @NotNull SelfUser bot) {
+        String reason = "User posted scam that was automatically detected";
+
+        actionsStore.addAction(guild.getIdLong(), bot.getIdLong(), author.getIdLong(),
+                ModerationAction.QUARANTINE, null, reason);
+
+        guild
+            .addRoleToMember(author,
+                    ModerationUtils.getQuarantinedRole(guild, config).orElseThrow())
+            .reason(reason)
+            .queue();
+    }
+
+    private void reportScamMessage(@NotNull GuildMessageReceivedEvent event,
+            @NotNull String reportTitle, @Nullable ActionRow confirmDialog) {
+        Guild guild = event.getGuild();
+        Optional<TextChannel> reportChannel = getReportChannel(guild);
+        if (reportChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to report a scam message, did not find a report channel matching the configured pattern '{}' for guild '{}'",
+                    reportChannelPattern, guild.getName());
+            return;
+        }
+
+        User author = event.getAuthor();
+        MessageEmbed embed =
+                new EmbedBuilder().setDescription(event.getMessage().getContentStripped())
+                    .setTitle(reportTitle)
+                    .setAuthor(author.getAsTag(), null, author.getAvatarUrl())
+                    .setTimestamp(event.getMessage().getTimeCreated())
+                    .setColor(AMBIENT_COLOR)
+                    .setFooter(author.getId())
+                    .build();
+        Message message =
+                new MessageBuilder().setEmbeds(embed).setActionRows(confirmDialog).build();
```

Comment:
Don't really see why you would want this?
Using the return value of sendMessage would be nicer I'd say?

Reply:
i dont follow, what do u mean by that?

Reply:
See below